### PR TITLE
Avalanche changes

### PR DESCRIPTION
**Pull request by piyush2-0**

### CODE REVIEW

File: contracts/aggregator/avalanche/flashloan/variables.sol [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 
-import {IndexInterface, ListInterface, IAaveLending} from "./interfaces.sol";
+import {IndexInterface, ListInterface, IAaveLending, Comptroller, IERC3156FlashLender} from "./interfaces.sol";
 
 contract ConstantVariables {
     address public constant aaveLendingAddr =
         0x4F01AeD16D97E3aB5ab2B501154DC9bb0F1A5A2C;
     IAaveLending public constant aaveLending = IAaveLending(aaveLendingAddr);
 
+    address public constant comptroller = 0x2eE80614Ccbc5e28654324a66A396458Fa5cD7Cc;
```
Comment (Vaibhav20201) [inline contracts/aggregator/avalanche/flashloan/variables.sol:11]:
```suggestion
    address public constant creamComptroller = 0x2eE80614Ccbc5e28654324a66A396458Fa5cD7Cc;
```

**Review by Vaibhav20201 — CHANGES_REQUESTED** [review]

File: contracts/aggregator/avalanche/flashloan/helpers.sol [hunk 2]
```diff
@@ -141,7 +141,9 @@ contract Helper is Variables {
     {
         if (_route == 1) {
             BPS_ = aaveLending.FLASHLOAN_PREMIUM_TOTAL();
-        } else {
+        } else if (_route == 8){
+            BPS_ = 3;
```
[the remark this answers is not in the document]
Comment (piyush2-0) [reply contracts/aggregator/avalanche/flashloan/helpers.sol:145]: I couldn't find a function to get from source, so I used it this way.

File: contracts/aggregator/avalanche/flashloan/main.sol [hunk 3]
```diff
@@ -11,9 +11,41 @@ import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
...
+        tokenToCrToken[usdc] = _crTokens[3];
+
+        address dai = address(0xd586E7F844cEa2F87f50152665BCbc2C279D8d70);
+        tokenToCrToken[dai] = _crTokens[4];
+
+        address wbtc = address(0x50b7545627a5162F82A992c33b87aDc75187B218);
+        tokenToCrToken[wbtc] = _crTokens[5];
+
+        address link = address(0x5947BB275c521040051D82396192181b413227A3);
+        tokenToCrToken[link] = _crTokens[6];
```
[the remark this answers is not in the document]
Comment (piyush2-0) [reply contracts/aggregator/avalanche/flashloan/main.sol:44]: Actually, the .underlying function which is used to get the underlying crToken is not present in CREAM, so I had to use it this way.

File: contracts/aggregator/avalanche/flashloan/main.sol [hunk 4]
```diff
@@ -88,6 +120,106 @@ contract FlashAggregatorAvalanche is Helper {
...
+        // );
+        // require(
+        //     borrowToken == a._token,
+        //     "encoded data (borrowToken) does not match"
+        // );
+        // require(
+        //     borrowAmount == a._amount,
+        //     "encoded data (borrowAmount) does not match"
+        // );
+        IERC20(data._token).approve(msg.sender, data._amount + data._fee);
```
[the remark this answers is not in the document]
Comment (piyush2-0) [reply contracts/aggregator/avalanche/flashloan/main.sol:194]: Okay

File: contracts/aggregator/avalanche/flashloan/main.sol [hunk 5]
```diff
@@ -88,6 +120,106 @@ contract FlashAggregatorAvalanche is Helper {
...
+
+        FlashloanVariables memory instaLoanVariables_;
+
+        (
+            uint256 route_,
+            address[] memory tokens_,
+            uint256[] memory amounts_,
+            address sender_,
+            bytes memory data_
+        ) = abi.decode(_data, (uint256, address[], uint256[], address, bytes));
```
[the remark this answers is not in the document]
Comment (piyush2-0) [reply contracts/aggregator/avalanche/flashloan/main.sol:161]: The docs suggested to encode all things, so i encoded everything.

**Pull request closed without merging**